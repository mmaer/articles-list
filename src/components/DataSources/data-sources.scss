@import '../../styles/variable.scss';

.data-sources {
  &__sources {
    display: flex;
    margin: 5px 0;

    @media (#{$above-tablet}) {
      flex-direction: column;
    }
  }

  &__source {
    margin-right: 5px;
  }
}
