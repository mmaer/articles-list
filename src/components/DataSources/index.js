export { default } from './DataSources.jsx';
