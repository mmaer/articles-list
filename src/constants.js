export const URLS = ['http://localhost:6010/articles/fashion', 'http://localhost:6010/articles/sports'];

export const SOURCES = ['fashion', 'sport'];
