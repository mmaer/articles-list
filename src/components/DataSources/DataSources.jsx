import React from 'react';

import './data-sources.scss';

const DataSources = ({ setSource, sources }) => {
  const onChange = ({ target }) => {
    const sourceIndex = sources.indexOf(target.name);
    if (sourceIndex === -1) {
      setSource([...sources, target.name]);
    } else {
      setSource([...sources.slice(0, sourceIndex), ...sources.slice(sourceIndex + 1)]);
    }
  };

  return (
    <div className="data-sources">
      <div className="data-sources__text">Data sources:</div>
      <div className="data-sources__sources">
        <div className="data-sources__source">
          <label>
            <input name="fashion" type="checkbox" checked={sources.includes('fashion')} onChange={onChange} />
            Fashion
          </label>
        </div>
        <div className="data-sources__source">
          <label>
            <input name="sport" type="checkbox" checked={sources.includes('sport')} onChange={onChange} />
            Sports
          </label>
        </div>
      </div>
    </div>
  );
};

export default DataSources;
