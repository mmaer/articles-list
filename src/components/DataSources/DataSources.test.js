import React from 'react';
import { shallow, mount } from 'enzyme';

import { SOURCES } from '../../constants';

import DataSources from './';

describe('DataSources component', () => {
  it('Should match snapshot', () => {
    const wrapper = shallow(<DataSources sources={SOURCES} setSource={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call setSource function when checkbox is clicked', () => {
    const mockSetSource = jest.fn();
    const wrapper = mount(<DataSources sources={SOURCES} setSource={mockSetSource} />);
    const fashionInput = wrapper.find('input[name="fashion"]');
    fashionInput.simulate('change', { target: { name: SOURCES[0] } });

    expect(mockSetSource.mock.calls.length).toBe(1);
  });

  it('Should remove one source when checkbox is clicked', () => {
    const mockSetSource = jest.fn();
    const wrapper = mount(<DataSources sources={SOURCES} setSource={mockSetSource} />);
    const fashionInput = wrapper.find('input[name="fashion"]');
    fashionInput.simulate('change', { target: { name: SOURCES[0] } });

    expect(mockSetSource.mock.calls[0]).toEqual([[SOURCES[1]]]);
  });

  it('Should add one source when checkbox is clicked', () => {
    const mockSetSource = jest.fn();
    const wrapper = mount(<DataSources sources={[SOURCES[0]]} setSource={mockSetSource} />);
    const fashionInput = wrapper.find('input[name="fashion"]');
    fashionInput.simulate('change', { target: { name: SOURCES[1] } });

    expect(mockSetSource.mock.calls[0]).toEqual([SOURCES]);
  });
});
