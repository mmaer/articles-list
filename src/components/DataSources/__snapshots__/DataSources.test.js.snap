// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSources component Should match snapshot 1`] = `
<div
  className="data-sources"
>
  <div
    className="data-sources__text"
  >
    Data sources:
  </div>
  <div
    className="data-sources__sources"
  >
    <div
      className="data-sources__source"
    >
      <label>
        <input
          checked={true}
          name="fashion"
          onChange={[Function]}
          type="checkbox"
        />
        Fashion
      </label>
    </div>
    <div
      className="data-sources__source"
    >
      <label>
        <input
          checked={true}
          name="sport"
          onChange={[Function]}
          type="checkbox"
        />
        Sports
      </label>
    </div>
  </div>
</div>
`;
